### PR TITLE
GG-18362: Move exception messages to property file

### PR DESCRIPTION
**Pull request by smoldachev**

### CODE REVIEW

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ConfigurationsRepository.java [hunk 1]
```diff
@@ -75,20 +83,38 @@
...
-            modelsIdx = new OneToManyIndex<>(ignite, "wc_cluster_models_idx");
-
-            clustersIdx = new OneToManyIndex<>(ignite, "wc_account_clusters_idx");
-            cfgIdx = new OneToManyIndex<>(ignite, "wc_account_configs_idx");
+            cachesIdx = new OneToManyIndex<>(
+                    ignite,
+                    "wc_cluster_caches_idx",
+                    (key) -> this.messages.getMessage(ERR_DATA_ACCESS_VIOLATION)
+            );
+            modelsIdx = new OneToManyIndex<>(
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ConfigurationsRepository.java]: Lets  add empty line.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ConfigurationsRepository.java [hunk 2]
```diff
@@ -75,20 +83,38 @@
...
+                    "wc_cluster_models_idx",
+                    (key) -> this.messages.getMessage(ERR_DATA_ACCESS_VIOLATION)
+            );
+
+            clustersIdx = new OneToManyIndex<>(
+                    ignite,
+                    "wc_account_clusters_idx",
+                    (key) -> this.messages.getMessage(ERR_DATA_ACCESS_VIOLATION)
+            );
+            cfgIdx = new OneToManyIndex<>(
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ConfigurationsRepository.java]: Lets  add empty line.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsHandler.java [hunk 3]
```diff
@@ -81,7 +89,7 @@ private void validateAgentHandshake(AgentHandshakeRequest req) {
         Collection<Account> accounts = accRepo.getAllByTokens(tokens);
 
         if (accounts.isEmpty())
-            throw new IllegalArgumentException("Failed to authenticate with token(s): " + tokens);
+            throw new IllegalArgumentException(messages.getMessage(ERR_FAILED_AUTH_WITH_TOKENS, tokens.toArray()));
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsHandler.java]: May be we can have method that also accepts collections?

File: modules/web-console/web-console-server/src/main/resources/i18/errors.properties [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+err.db-lost-connection-during-tx=Database connection was lost during transaction
+err.active-tx-not-found=No active transaction was found
+err.data-access-violation=Data access violation
+err.db-not-available=Database not available
+
+err.prohibited-revoke-admin-rights=Self revoke of administrator rights is prohibited
+err.parse-signin-req-failed=Failed to parse signin request
+err.could-not-create-hash=Could not create hash
+err.tokens-no-specified-in-agent-handshake-req=Tokens not specified in agent handshake request
+err.failed-auth-with-tokens=Failed to authenticate with token(s): {0}
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/resources/i18/errors.properties]: Do we need indexed placeholders?

File: modules/web-console/web-console-server/src/main/resources/i18/errors.properties [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+err.model-id-not-found=Model ID not found
+
+err.cache-not-found-by-id=Cache not found for ID: {0}
+err.cache-id-not-found=Cache ID not found
+
+err.account-not-found-by-token=Failed to find account with this token! Please check link from email.
+err.account-not-found-by-id=Account not found for ID: {0}
+err.account-with-email-exists=Account with email {0} already registered
+err.account-with-token-exists=Account with token {0} already exists
+err.account-cant-be-found-in-ws-session=Account can't be found [session={0}]
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/resources/i18/errors.properties]: Empty line in the end of file.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/errors/Errors.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+package org.apache.ignite.console.errors;
+
+import javax.cache.CacheException;
+import org.apache.ignite.IgniteClientDisconnectedException;
+import org.apache.ignite.IgniteException;
+
+/**
+ * Class with error codes and messages.
+ */
+public class Errors {
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/errors/Errors.java]: Why we need string constant? I can't find same pattern in spring library

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/config/ApplicationConfiguration.java [hunk 7]
```diff
@@ -33,15 +34,17 @@
 @EnableScheduling
 public class ApplicationConfiguration {
     /**
+     * @param messages Messages accessor.
+     *
      * @return Service for encoding user passwords.
      */
     @Bean
-    public PasswordEncoder passwordEncoder() {
+    public PasswordEncoder passwordEncoder(MessageSourceAccessor messages) {
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/config/ApplicationConfiguration.java]: Why not create MessageSource like here: org.springframework.security.core.SpringSecurityMessageSource

File: modules/web-console/web-console-server/src/main/resources/i18/errors.properties (hunk 5 above)
Comment (smoldachev) [reply modules/web-console/web-console-server/src/main/resources/i18/errors.properties]: Fix.

File: modules/web-console/web-console-server/src/main/resources/i18/errors.properties (hunk 4 above)
Comment (smoldachev) [reply modules/web-console/web-console-server/src/main/resources/i18/errors.properties]: Spring MessageSource using https://docs.oracle.com/javase/7/docs/api/java/text/MessageFormat.html under the hood.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ConfigurationsRepository.java (hunk 2 above)
Comment (smoldachev) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ConfigurationsRepository.java]: Fix.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ConfigurationsRepository.java (hunk 1 above)
Comment (smoldachev) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/ConfigurationsRepository.java]: Fix.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/errors/Errors.java (hunk 6 above)
Comment (smoldachev) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/errors/Errors.java]: String constant better for refactoring cases, and we can find all usages in one click.

File: modules/web-console/web-console-mail/src/test/resources/i18/notifications.properties [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+notifications.simple.subject=subject
+notifications.simple.body=text
+
+notifications.spel.subject=Hello ${recipient.firstName} ${recipient.lastName}! subject
+notifications.spel.body=text
```
[the remark this answers is not in the document]
Comment (smoldachev) [reply modules/web-console/web-console-mail/src/test/resources/i18/notifications.properties]: Fix.

File: modules/web-console/web-console-common/pom.xml [hunk 9]
```diff
@@ -59,6 +59,20 @@
...
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>${spring.version}</version>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+            <version>3.9</version>
```
[the remark this answers is not in the document]
Comment (smoldachev) [reply modules/web-console/web-console-common/pom.xml]: Move version to parent.
I think yes, work with strings in java is pain :(

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/errors/Errors.java (hunk 6 above)
Comment (nva) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/errors/Errors.java]: In IDEA you can click on message code in properties file and receive all usage. If you need to change message code you should change const then change message code it's very strange.

Comment (smoldachev) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/errors/Errors.java]: Well, let's consider the following case:
We change the message code in the properties file.

Without constants, we must change each file with a changed message code.
With constants, we have to change one place and not worry about missing changes in other files.

With or without constants, we have to change more than one place, and I think one change is better than a change in many files.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsHandler.java (hunk 3 above)
Comment (smoldachev) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsHandler.java]: Fix in WebConsoleMessageSourceAccessor.

File: modules/web-console/web-console-server/src/test/java/org/apache/ignite/console/MockTestConfiguration.java [hunk 10]
```diff
@@ -43,9 +43,9 @@
 /**
  * Test configuration with mocks.
  */
-@Configuration
+@TestConfiguration
 @Import(Application.class)
-public class TestConfiguration {
+public class MockTestConfiguration {
```
[the remark this answers is not in the document]
Comment (smoldachev) [reply modules/web-console/web-console-server/src/test/java/org/apache/ignite/console/MockTestConfiguration.java]: Done.

**Pull request merged**